core/chains/evm/client/rpc_client.go

```diff
@@ -396,8 +401,26 @@ func (r *rpcClient) CallContext(ctx context.Context, result interface{}, method
 	return err
 }
 
-func (r *rpcClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) error {
-	ctx, cancel, ws, http := r.makeLiveQueryCtxAndSafeGetClients(ctx, r.largePayloadRpcTimeout)
+func (r *rpcClient) BatchCallContext(rootCtx context.Context, b []rpc.BatchElem) error {
```

Comment:
Is it intentional that we aren't calling "onNewFinalizedHead()" in this call, even if it had an element requesting Final block?

Reply:
Yes, it is intentional. BatchCallContext is not used by HeadTracker to fetch latest finalized block.
We discourage usage of direct RPC calls to identify latest finalized block, so there is no need for us to intercept it.